source/agora/consensus/EnrollmentManager.d

```diff
@@ -1142,3 +1158,62 @@ unittest
     man.getEnrollments(enrolls, Height(validator_cycle + 11));
     assert(enrolls.length == 0);
 }
+
+/// Test for restoring enrollment data and pre-image information for a node
+unittest
+{
+    import agora.common.Amount;
+    import agora.consensus.data.Transaction;
+    import agora.consensus.Genesis;
+    import std.conv;
+
+    scope utxo_set = new TestUTXOSet;
+    auto gen_key_pair = WK.Keys.Genesis;
+    auto key_pair = KeyPair.random();
+
+    foreach (uint idx; 0 .. 7)
+    {
+        auto tx = TxBuilder(GenesisTransaction, idx).
+            draw(Amount(Amount.MinFreezeAmount), [gen_key_pair.address]).
+            sign(TxType.Freeze);
+        utxo_set.put(tx);
+    }
+    auto finder = &utxo_set.findUTXO;
+    auto utxos = utxo_set.storage;
+
+    // create enrollments for nodes
+    auto params = new immutable(ConsensusParams)();
+    auto man = new EnrollmentManager(":memory:", gen_key_pair, params);
```

Comment:
We should stop using the genesis keypair for those tests, it's a weird edge case that is no realistic.